main.go

```diff
@@ -175,16 +179,31 @@ func startExtAuthServerGRPC(serviceCache cache.Cache) {
 		logger.Error(err, "failed to obtain port for grpc auth service")
 		os.Exit(1)
 	} else {
-		opts := []grpc.ServerOption{grpc.MaxConcurrentStreams(gRPCMaxConcurrentStreams)}
-		s := grpc.NewServer(opts...)
+		grpcServerOpts := []grpc.ServerOption{grpc.MaxConcurrentStreams(gRPCMaxConcurrentStreams)}
 
-		envoy_auth.RegisterAuthorizationServer(s, &service.AuthService{Cache: serviceCache})
-		healthpb.RegisterHealthServer(s, &service.HealthService{})
+		tlsEnabled := tlsCertPath != "" && tlsCertKeyPath != ""
+		logger.Info("starting grpc service", "port", extAuthGRPCPort, "tls", tlsEnabled)
 
-		logger.Info("starting grpc service", "port", extAuthGRPCPort)
+		if tlsEnabled {
+			if tlsCert, err := tls.LoadX509KeyPair(tlsCertPath, tlsCertKeyPath); err != nil {
```

Comment:
We're going with `crypto/tls` here instead of using `openssl` simply because that's what Golang's HTTP server uses and we are already using it for the OIDC wristband validation endpoints. So no two ways of doing the same thing.